packages/arb-token-bridge-ui/src/token-bridge-sdk/EnhancedProvider.ts

```diff
@@ -130,16 +134,62 @@ function addTxReceiptToCache(
   )
 }
 
-export class EnhancedProvider extends StaticJsonRpcProvider {
+export class EnhancedProvider extends JsonRpcBatchProvider {
+  // `detectNetwork()` to give this provider StaticJsonRpcProvider's functionality
+  // copied from https://github.com/ethers-io/ethers.js/blob/v5/packages/providers/src.ts/url-json-rpc-provider.ts#L28
+  async detectNetwork(): Promise<Network> {
+    let network = this.network
+    if (network == null) {
+      network = await super.detectNetwork()
+
+      if (!network) {
+        logger.throwError(
+          'no network detected',
+          Logger.errors.UNKNOWN_ERROR,
+          {}
+        )
+      }
+
+      // If still not set, set it
+      if (this._network == null) {
+        // A static network does not support "any"
+        defineReadOnly(this, '_network', network)
+
+        this.emit('network', network, null)
+      }
+    }
+    return network
+  }
+
   private storage: Storage
+  private options: ProviderOptions
+  private staticProvider?: StaticJsonRpcProvider
 
   constructor(
     url?: ConnectionInfo | string,
     network?: Networkish,
-    storage: Storage = new WebStorage()
+    storage: Storage = new WebStorage(),
+    options: ProviderOptions = {
+      enableCaching: true,
+      enableBatching: true
+    }
   ) {
```

Comment:
i see that `storage` is `undefined` when initialised, but we still pass in `options`, it's better to use an object here in the constructor than multiple args